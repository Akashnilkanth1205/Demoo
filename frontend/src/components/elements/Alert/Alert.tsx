/**
 * @license
 * Copyright 2018-2020 Streamlit Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { Alert as AlertProto, IAlert } from "autogen/proto"
import AlertContainer, { Kind } from "components/shared/AlertContainer"
import { StreamlitMarkdown } from "components/shared/StreamlitMarkdown"
import { requireNonNull } from "lib/utils"
import React, { ReactElement } from "react"

export function getAlertKind(format: AlertProto.Format): Kind {
  switch (format) {
    case AlertProto.Format.ERROR:
      return Kind.ERROR
    case AlertProto.Format.INFO:
      return Kind.INFO
    case AlertProto.Format.SUCCESS:
      return Kind.SUCCESS
    case AlertProto.Format.WARNING:
      return Kind.WARNING
    default:
      throw new Error(`Unexpected alert type: ${format}`)
  }
}

export interface AlertProps {
  width: number
  element: IAlert
}

/**
 * Functional element representing error/warning/info/success boxes
 * which may be formatted in Markdown.
 */
export default function Alert({ element, width }: AlertProps): ReactElement {
  const body = requireNonNull(element.body)
  const format = requireNonNull(element.format)

  return (
    <div className="stAlert">
      <AlertContainer width={width} kind={getAlertKind(format)}>
        <div className="markdown-text-container">
          <StreamlitMarkdown source={body} allowHTML={false} />
        </div>
      </AlertContainer>
    </div>
  )
}
