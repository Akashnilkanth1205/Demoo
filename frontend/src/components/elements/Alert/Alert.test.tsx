/**
 * @license
 * Copyright 2018-2019 Streamlit Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react"
import ReactDOM from "react-dom"
import { Map as ImmutableMap } from "immutable"
import Alert from "./Alert"
import renderer from "react-test-renderer"

const getProps = (elementProps: object = {}): Props => ({
  element: ImmutableMap({
    body: "Something happened!",
    ...elementProps,
  }),
  width: 100,
})

describe("Alert Element Test", () => {
  it("renders an ERROR box", () => {
    const props = getProps()
    const component = renderer.create(<Alert format="error" {...props} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders an INFO box", () => {
    const props = getProps()
    const component = renderer.create(<Alert format="info" {...props} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders a WARNING box", () => {
    const props = getProps()
    const component = renderer.create(<Alert format="warning" {...props} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders a SUCCESS box", () => {
    const props = getProps()
    const component = renderer.create(<Alert format="success" {...props} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
