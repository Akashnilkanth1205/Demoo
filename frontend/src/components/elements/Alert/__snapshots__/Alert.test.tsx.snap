// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Element Test renders a SUCCESS box 1`] = `
<div
  className="alert stText"
  style={
    Object {
      "width": 100,
    }
  }
>
  <div
    className="markdown-text-container"
  >
    <p>
      Something happened!
    </p>
  </div>
</div>
`;

exports[`Alert Element Test renders a WARNING box 1`] = `
<div
  className="alert stText"
  style={
    Object {
      "width": 100,
    }
  }
>
  <div
    className="markdown-text-container"
  >
    <p>
      Something happened!
    </p>
  </div>
</div>
`;

exports[`Alert Element Test renders an ERROR box 1`] = `
<div
  className="alert stText"
  style={
    Object {
      "width": 100,
    }
  }
>
  <div
    className="markdown-text-container"
  >
    <p>
      Something happened!
    </p>
  </div>
</div>
`;

exports[`Alert Element Test renders an INFO box 1`] = `
<div
  className="alert stText"
  style={
    Object {
      "width": 100,
    }
  }
>
  <div
    className="markdown-text-container"
  >
    <p>
      Something happened!
    </p>
  </div>
</div>
`;
